src/app/components/TopicForm/TopicForm.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import type { Topic } from '../../types/types'
+import Button from '../Button/Button'
+
+type TopicFormProps = {
+  onSubmit: (newTopic: Topic) => void
+  onCancel: () => void
+}
+
+const MAX_TITLE_LENGTH = 40
+const MAX_DESCRIPTION_LENGTH = 144
+
+export default function TopicForm({
+  onSubmit,
+  onCancel,
+}: TopicFormProps): JSX.Element {
+  const [title, setTitle] = useState('')
+  const [description, setDescription] = useState('')
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    const newTopic = {
+      id: Math.floor(Math.random() * 10000),
```

Comment:
I'd recommend a package for that, although you might want to wait implementing one until you get to the backend phase, if you plan to add a backend.
https://github.com/ai/nanoid